boards/seeed/xiao_esp32c6/doc/index.rst

```diff
@@ -0,0 +1,232 @@
+.. zephyr:board:: xiao_esp32c6
+
+Overview
+********
+
+Seeed Studio XIAO ESP32C6 is powered by the highly-integrated ESP32-C6 SoC.
+It consists of a high-performance (HP) 32-bit RISC-V processor, which can be clocked up to 160 MHz,
+and a low-power (LP) 32-bit RISC-V processor, which can be clocked up to 20 MHz.
+It has a 320KB ROM, a 512KB SRAM, and works with external flash.
+This board integrates complete Wi-Fi, Bluetooth LE, Zigbee, and Thread functions.
+For more information, check `Seeed Studio XIAO ESP32C6`_ .
+
+Hardware
+********
+
+This board is based on the ESP32-C6 with 4MB of flash, integrating 2.4 GHz Wi-Fi 6,
+Bluetooth 5.3 (LE) and the 802.15.4 protocol. It has an USB-C port for programming
+and debugging, integrated battery charging and an U.FL external antenna connector.
+It is based on a standard XIAO 14 pin pinout.
+
+Supported Features
+==================
+
+Current Zephyr's XIAO ESP32C6 board supports the following features:
```

Comment:
Replace with board name as used by west with \`\` around it and refer to it as board target